src/Mvc/Mvc.Core/src/Filters/FilterFactory.cs

```diff
@@ -78,6 +87,11 @@ internal static class FilterFactory
                 throw new ArgumentNullException(nameof(cachedFilterItems));
             }
 
+            if (actionContext.ActionDescriptor.CachedResuableFilters is { } cached)
```

Comment:
I dislike this pattern and I also dislike that `is not` doesn't let you capture the thing that isn't null